### ISSUE TITLE
proxy tool doesn't work with Azure headers

### ISSUE THREAD

**Issue by jaraco** [cherrypy/cherrypy#1626]
* **I'm submitting a ...**
[X ] bug report
[ ] feature request
[ ] question about the decisions made in the repository

* **Do you want to request a *feature* or report a *bug*?**

Yes

* **What is the current behavior?**

When behind an SSL reverse proxy in Azure, it supplies a header like this one:

`"X-Arr-Ssl": "2048|256|C=US, S=Washington, L=Redmond, O=Microsoft Corporation, OU=Microsoft IT, CN=Microsoft IT SSL SHA2|CN=*.azurewebsites.net"`

The proxy tool has support for supplying the HTTPS scheme if the appropriate header is sent, but no options to the proxy tool will allow it to detect SSL for the Azure header.

* **What is the expected behavior?**

The proxy tool should be extensible such that it supports many different hosts and not just webfaction.

* **What is the motivation / use case for changing the behavior?**

Results from `cherrypy.url()` are incorrect for HTTPS requests when hosted behind Azure.

* **Please tell us about your environment:**

- Python version: 3.6.2
- OS: Linux
- Browser: n/a

**Comment by webknjaz** [comment]
> Results from cherrypy.url() are incorrect for HTTPS requests when hosted behind Azure.

Same for Heroku and probably many others. Currently I solve this by providing additional environment variable with the base URL, but of course this has to be redesigned.

**Comment by jaraco** [comment]
Do you mean something like `HTTP_X_FORWARDED_PROTO=https`?

**Comment by jaraco** [comment]
As a workaround, I ended up creating this tool:

```python
"""
CherryPy Azure support
"""

import cherrypy


def azure_proxy():
	"""
	As described in cherrypy/cherrypy#1626, CherryPy can't detect
	SSL on Azure, so cajole it so.
	"""
	request = cherrypy.serving.request
	if request.headers.get('X-Arr-SSL'):
		request.headers['X-Forwarded-Proto'] = 'https'


cherrypy.tools.azure_proxy = cherrypy.Tool(
	'before_request_body',
	azure_proxy,
	priority=cherrypy.tools.proxy._priority - 1,
)
```

and enabling it thus:

```python
				'tools.proxy.on': True,
				'tools.proxy.remote': 'X-Client-IP',
				'tools.azure_proxy.on': True,
```

**Comment by webknjaz** [comment]
> Do you mean something like HTTP_X_FORWARDED_PROTO=https?

No, actually I just supply it via cherrypy config and then I've got lots of crutches around the app for url generation.
